src/main/java/com/thealgorithms/stacks/InfixToPrefix.java

```diff
@@ -0,0 +1,83 @@
+package com.thealgorithms.stacks;
+
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class InfixToPrefix {
+    private InfixToPrefix() {
+    }
+
+    /**
+     * Convert an infix expression to a prefix expression.
+     *
+     * @param infixExpression the infix expression to convert
+     * @return the prefix expression
+     * @throws Exception if the infix expression has unbalanced brackets
+     */
+    public static String infix2Prefix(String infixExpression) throws Exception {
+        if (!BalancedBrackets.isBalanced(filterBrackets(infixExpression))) {
+            throw new Exception("invalid expression");
```

Comment:
Will be better to use [IllegalArgumentException](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalArgumentException.html)

Reply:
Oh okay. Change implemented.